frontend/js/src/settings/brainzplayer/BrainzPlayerSettings.tsx

```diff
@@ -0,0 +1,216 @@
+import * as React from "react";
+import {
+  faSpotify,
+  faApple,
+  faSoundcloud,
+  faYoutube,
+} from "@fortawesome/free-brands-svg-icons";
+import { Helmet } from "react-helmet";
+import { Link } from "react-router-dom";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { toast } from "react-toastify";
+import Switch from "../../components/Switch";
+import GlobalAppContext from "../../utils/GlobalAppContext";
+import SpotifyPlayer from "../../common/brainzplayer/SpotifyPlayer";
+import SoundcloudPlayer from "../../common/brainzplayer/SoundcloudPlayer";
+import { ToastMsg } from "../../notifications/Notifications";
+
+function BrainzPlayerSettings() {
+  const {
+    spotifyAuth,
+    soundcloudAuth,
+    userPreferences,
+    APIService,
+    currentUser,
+  } = React.useContext(GlobalAppContext);
+  const [youtubeEnabled, setYoutubeEnabled] = React.useState(
+    userPreferences?.brainzplayer?.youtubeEnabled ?? true
+  );
+  const [spotifyEnabled, setSpotifyEnabled] = React.useState(
+    userPreferences?.brainzplayer?.spotifyEnabled ??
+      SpotifyPlayer.hasPermissions(spotifyAuth)
+  );
+  const [soundcloudEnabled, setSoundcloudEnabled] = React.useState(
+    userPreferences?.brainzplayer?.soundcloudEnabled ??
+      SoundcloudPlayer.hasPermissions(soundcloudAuth)
+  );
+
+  const saveSettings = React.useCallback(async () => {
+    if (!currentUser?.auth_token) {
+      toast.error("You must be logged in to update your preferences");
+      return;
+    }
+    const { submitBrainzplayerPreferences } = APIService;
+    try {
+      await submitBrainzplayerPreferences(currentUser.auth_token, {
+        youtubeEnabled,
+        spotifyEnabled,
+        soundcloudEnabled,
+      });
+      toast.success("Saved your preferences successfully");
+    } catch (error) {
+      toast.error(
+        <ToastMsg
+          title="Error saving preferences"
+          message={
+            <>
+              {error.toString()}
+              <br />
+              Please try again or contact us if the issue persists.
+            </>
+          }
+        />
+      );
```

Comment:
After the settings are updated, the context also has to be updated, otherwise the previous settings will be displayed to the user. This is because since this is a single page application, so the context is fetched only once.

To test it out, update the settings, navigate to the import listens page, and then navigate back to this route.

Reply:
Hah, I thought we had no page in the settings router where I could test it, forgot about that page XD